### ISSUE TITLE
IDISP0004 false positive for DiagnosticSource.StartActivity

### ISSUE THREAD

**Issue by Bouke** [DotNetAnalyzers/IDisposableAnalyzers#484]
[DiagnosticSource.StartActivity](https://learn.microsoft.com/en-us/dotnet/api/System.Diagnostics.DiagnosticSource.StartActivity?view=net-6.0) returns the `activity` instance that was passed as an argument. The analyzer suggests to also dispose the returned activity, however it is already being disposed with the using on the first line:

```cs
using var activity = new Activity("Do work");
activity.SetParentId(parentId);
↓diagnosticSource.StartActivity(activity, new { work });
```

However with a pattern like the following, we should still raise IDISP0004 as expected:

```cs
diagnosticSource.StartActivity(↓new Activity("Do work"), new { work });
```
